src/Aspire.Hosting/ApplicationModel/DeploymentTargetAnnotation.cs

```diff
@@ -12,4 +12,15 @@ public sealed class DeploymentTargetAnnotation(IResource target) : IResourceAnno
     /// The deployment target.
     /// </summary>
     public IResource DeploymentTarget { get; } = target;
+
+    /// <summary>
+    /// Gets or sets the container registry information associated with
+    /// the deployment target, if the deployment target an image-based environment.
+    /// </summary>
+    /// <remarks>
+    /// This property is typed as <see cref="object"/> since the
+    /// IContainerRegistry interface is defined in the layer above
+    /// in the Aspire.Hosting.Azure.AppContainers assemblies.
+    /// </remarks>
+    public object? ContainerRegistryInfo { get; set; }
```

Comment:
```suggestion
    public IContainerRegistry? ContainerRegistryInfo { get; set; }
```

Reply:
`IContainerRegistry` is defined in the AppContainer's assembly and we don't have access to it here so I hacked around this by using the `object` type. I don't think we'd want to define the container registry in `Aspire.Hosting`.

One thought I had was making this `DeploymentTargetAnnotatation` an abstract base class and having a `ContainerDeploymentTargetAnnotation` type in AppContainer's assembly that includes the `IContainerRegistry` property. That doesn't avoid the problem of needing to include AppContainer's references if you're trying to resolve the container registry.

Maybe we can also consider creating a new assembly for deployment related concepts? `Aspire.Deployment`?